lumibot/brokers/broker.py

```diff
@@ -1098,3 +1111,17 @@ def export_trade_events_to_csv(self, filename):
         if len(self._trade_event_log_df) > 0:
             output_df = self._trade_event_log_df.set_index("time")
             output_df.to_csv(filename)
+
+    def set_strategy_name(self, strategy_name):
+        """
+        Let's the broker know the name of the strategy that is using it for logging purposes.
+
+        Parameters
+        ----------
+        strategy_name : str
+            The name of the strategy that is using the broker.
+        """
+        self._strategy_name = strategy_name
+
+        # Create an adapter with 'strategy_name' set to the instance's name
+        self.logger = CustomLoggerAdapter(logger, {'strategy_name': self._strategy_name})
```

Comment:
I think it is better to have a function inside of the CustomLoggerAdapter that just updates the strategy name being used rather than creating an entirely new object.

Reply:
fixed. pr coming